rts/lra/API.adoc

```diff
@@ -0,0 +1,85 @@
+= Versioning of Narayna LRA REST API
+
+The goal of this document is summarize the approach to REST API versioning
+in Narayana LRA coordinator service.
+This document is written for developer of Narayana LRA services.
+
+The version format is `major.minor`[`-preRelease`].
+The `major` and `minor` parts are required and `-preRelease` is used
+during development and is optional.
+
+NOTE: Any final `minor.major` release is considered higher than
+      the same version with `-preRelease` part.
+
+Client may demand behaviour based on particular API version
+by providing HTTP header link:./service-base/src/main/java/io/narayana/lra/LRAConstants.java[`Narayana-LRA-API-version`] on the call.
+
+== API definition as Open API document
+
+The Narayana LRA API is documented with Open API annotations at the java
+classes. The Open API definition needs to be published at the http://narayana.io
+page.
+
+== Changes in LRA REST API
+
+The Narayana LRA REST API is expected to support for at least two previous
+`major` versions (ie. support is expected in parallel at least of versions 1.x,
+2.x and 3.x until the 4.0 is released).
```

Comment:
I understand it. My perspective would be probably to go with 1 version backwards in parallel instead of 2 but on the other hand @mmusgrov has valid points to maybe prolong the support even longer.
There are different perspectives. Maybe the requirements could be adjusted based on changes happening in the code and the feedback from the community.